mmdet/apis/inference.py

```diff
@@ -201,3 +204,64 @@ async def async_inference_detector(model, imgs):
     torch.set_grad_enabled(False)
     results = await model.aforward_test(data, rescale=True)
     return results
+
+
+def show_result_pyplot(model: BaseDetector,
+                       img: Union[str, np.ndarray],
+                       result: DetDataSample,
+                       opacity: float = 0.5,
+                       title: str = '',
+                       draw_gt: bool = True,
+                       draw_pred: bool = True,
+                       wait_time: float = 0,
+                       show: bool = True,
+                       save_dir=None,
+                       out_file=None):
+    """Visualize the detection results on the image.
+
+    Args:
+        model (BaseDetector): The loaded detector.
+        img (str or np.ndarray): Image filename or loaded image.
+        result (DetDataSample): The prediction DetDataSample result.
+        opacity (float): Opacity of painted detection map.
+            Default 0.5. Must be in (0, 1] range.
+        title (str): The title of pyplot figure.
+            Default is ''.
+        draw_gt (bool): Whether to draw GT DetDataSample. Default to True.
+        draw_pred (bool): Whether to draw Prediction DetDataSample.
+            Defaults to True.
+        wait_time (float): The interval of show (s). 0 is the special value
+            that means "forever". Defaults to 0.
+        show (bool): Whether to display the drawn image.
+            Default to True.
+        out_file (str, optional): Path to output file. Default to None.
+
+    Returns:
+        np.ndarray: the drawn image which channel is RGB.
+    """
+    if hasattr(model, 'module'):
+        model = model.module
+    if isinstance(img, str):
+        image = imread(img)
+    else:
+        image = img.copy()
+
+    # init visualizer
+    visualizer = DetLocalVisualizer(
```

Comment:
This is not universal and cannot be used with other visualizers.

Reply:
How can I  modify it better, or Is there any way to visualizer the results more simpler APi than visualizer.